code/modules/food_and_drinks/restaurant/customers/_customer.dm

```diff
@@ -155,6 +171,101 @@
+	base_icon = "mothbot"
+	found_seat_lines = list("Give me your hat!", "Moth?", "Certainly an... interesting venue.")
+	cant_find_seat_lines = list("If I can't find a seat, I'm flappity flapping out of here quick!", "I'm trying to flutter here!")
+	leave_mad_lines = list("I'm telling all my moth friends to never come here!", "Zero star rating, even worse than that time I ate a mothball!","Closing down permanently would still be too good of a fate for this place.")
+	leave_happy_lines = list("I'd tip you my hat, but I ate it!", "I hope that wasn't a collectible!", "That was the greatest thing I ever ate, even better than Guanaco!")
+	wait_for_food_lines = list("How hard is it to get food here? You're even wearing food yourself!", "My fuzzy robotic tummy is rumbling!", "I don't like waiting!")
+	friendly_pull_line = "Moff?"
+	first_warning_line = "Go away, I'm trying to get some hats here!"
+	second_warning_line = "Last warning! I'll destroy you!"
+	self_defense_line = "Flap attack!"
+
+	speech_sound = 'sound/creatures/tourist/tourist_talk_moth.ogg'
+
+	// Always asks for the clothes that you have on, but this is a fallback.
+	orderable_objects = list(
+		/datum/venue/restaurant = list(
+			/obj/item/clothing/head/chefhat = 3,
+			/obj/item/clothing/shoes/sneakers/black = 3,
+			/obj/item/clothing/gloves/color/black = 1,
+		),
+	)
+
+	clothing_sets = list("mothbot_clothes")
+	is_unique = TRUE
+
+	/// The wings chosen for the moth customers.
+	var/list/wings_chosen
+
+// The whole gag is taking off your hat and giving it to the customer.
+// If it takes any more effort, it loses a bit of the comedy.
+// Therefore, only show up if it's reasonable for that gag to happen.
+/datum/customer_data/moth/can_use(datum/venue/venue)
+	return !isnull(get_dynamic_order(venue))
+
+/datum/customer_data/moth/proc/get_dynamic_order(datum/venue/venue)
+	var/mob/living/carbon/buffet = venue.restaurant_portal?.turned_on_portal?.resolve()
+	if (!istype(buffet))
+		return
+
+	var/list/orderable = list()
+
+	if (!QDELETED(buffet.head))
+		orderable[buffet.head] = 5
+
+	if (!QDELETED(buffet.gloves))
+		orderable[buffet.gloves] = 5
+
+	if (!QDELETED(buffet.shoes))
+		orderable[buffet.shoes] = 1
+
+	if (orderable.len)
+		var/datum/order = pickweight(orderable)
+		return order.type
+
+/datum/customer_data/moth/proc/get_wings(mob/living/simple_animal/robot_customer/customer)
+	var/customer_ref = WEAKREF(customer)
+	if (!LAZYACCESS(wings_chosen, customer_ref))
+		LAZYSET(wings_chosen, customer_ref, GLOB.moth_wings_list[pick(GLOB.moth_wings_list)])
+	return wings_chosen[customer_ref]
+
+/datum/customer_data/moth/get_underlays(mob/living/simple_animal/robot_customer/customer)
+	var/list/underlays = list()
+
+	var/datum/sprite_accessory/moth_wings/wings = get_wings(customer)
+
+	var/mutable_appearance/wings_behind = mutable_appearance(icon = 'icons/mob/moth_wings.dmi', icon_state = "m_moth_wings_[wings.icon_state]_BEHIND")
+	wings_behind.appearance_flags = RESET_COLOR
```

Comment:
is there any point in caching this?

Reply:
I'm not sure what you mean, sorry.

Reply:
well we're making the same mutable_appearance each time this proc is called, would it be worth caching?

Reply:
We're not.

This proc is only called once, and depends on the icon state of the wings.